Submitted By: Alexander E. Patrakov
Date: 2004-05-13
Initial Package Version: 2004_04_01
Upstream Status: will send a copy to linux-hotplug-devel@lists.sourceforge.net
Origin: Alexander E. Patrakov
Description: Removes bogus dependency upon "which" and "usbmodules"
programs in 2.6 code path

--- hotplug-2004_04_01/etc/hotplug/usb.rc.old	2004-05-13 20:42:25.596236912 +0600
+++ hotplug-2004_04_01/etc/hotplug/usb.rc	2004-05-13 20:43:09.296593448 +0600
@@ -73,12 +73,6 @@
 	#
 
 	# synthesize hotplug events if we can 
-	# we need (non-bash) programs to parse descriptors.
-	LISTER=`which usbmodules`
-	if [ "$LISTER" = "" -o ! -f /proc/bus/usb/devices ]; then
-		echo $"** can't synthesize root hub events"
-		return
-	fi
 
 	# make sure the usb agent will run
 	ACTION=add
@@ -113,6 +107,12 @@
 		done
 	    fi
 	else
+    	    # we need (non-bash) programs to parse descriptors.
+	    LISTER=`which usbmodules`
+	    if [ "$LISTER" = "" -o ! -f /proc/bus/usb/devices ]; then
+		echo $"** can't synthesize root hub events"
+		return
+	    fi
 	    for DEVICE in /proc/bus/usb/*/*; do
 		/etc/hotplug/usb.agent
 	    done

